test/unit/SectionField/Generator/PhpFormatterTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+declare (strict_types=1);
+
+namespace Tardigrades\SectionField\Generator;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass Tardigrades\SectionField\Generator\PhpFormatter
+ */
+final class PhpFormatterTest extends TestCase
+{
+    /**
+     * @test
+     * @covers ::format
+     */
+    public function it_should_format()
+    {
+        $inputText = <<<TXT
+<?php
+declare (strict_types=1);
+namespace {{ namespace }};
+use Tardigrades;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+use Symfony\Component\Validator\Constraints as Assert;
+{{ use }}
+class {{ section }}
+{
+    {{ properties }}
+    /** @var int */
+    private \$id;
+    /** @var string */
+    private \$name;
+    public function __construct()
+    {
+        {{ constructor }}
+    }
+
+    public function getId(): ?int
+    {
+        return \$this->id;
+    }
+
+    {{ methods }}
+    {{ getSlug }}
+    {{ getDefault }}
+
+    public static function loadValidatorMetadata(ClassMetadata \$metadata)
+    {
+        {{ validatorMetadata }}
+        {{ validatorMetadataSectionPhase }}
+    }
+
+    public function onPrePersist(): void
+    {
+        {{ prePersist }}
+    }
+
+    public function onPreUpdate(): void
+    {
+        {{ preUpdate }}
+    }
+}
+
+TXT;
+
+        $outputText = <<<TXT
+<?php
+declare (strict_types=1);
+
+namespace {{ namespace }};
+
+use Tardigrades;
+use Symfony\Component\Validator\Mapping\ClassMetadata;
+use Symfony\Component\Validator\Constraints as Assert;
+{{ use }}
+
+class {{ section }}
+{
+{{ properties }}
+    /** @var int */
+    private \$id;
+
+    /** @var string */
+    private \$name;
+
+    public function __construct()
+    {
+    {{ constructor }}
+    }
+
+    public function getId(): ?int
+    {
+        return \$this->id;
+    }
+{{ methods }}
+{{ getSlug }}
+{{ getDefault }}
+
+    public static function loadValidatorMetadata(ClassMetadata \$metadata)
+    {
+    {{ validatorMetadata }}
+    {{ validatorMetadataSectionPhase }}
+    }
+
+    public function onPrePersist(): void
+    {
+    {{ prePersist }}
+    }
+
+    public function onPreUpdate(): void
+    {
+    {{ preUpdate }}
+    }
+}
+
+
+TXT;
+
+        $result = PhpFormatter::format($inputText);
```

Comment:
Actually, the formatter itself never encounters variables (the psuedo variables like: {{ something }}). 

The process is: Generate methods, put them in the template (with the pseudo vars) and then start formatting.

So, to test this, some fake php stuff, wrongly formatted, would do it.

Reply:
Aha, clear. Using the real example code you provided, it works well. Thanks!